src/DIRAC/WorkloadManagementSystem/Agent/PilotLoggingAgent.py

```diff
@@ -80,36 +73,74 @@ def execute(self):
                     self.log.error(
                         f"No proxy user or group defined for pilot: VO: {vo}, User: {proxyUser}, Group: {proxyGroup}"
                     )
-                    voRes[vo] = "No proxy user or group defined - skipped"
                     continue
 
                 self.log.info(f"Proxy used for pilot logging: VO: {vo}, User: {proxyUser}, Group: {proxyGroup}")
-                res = self.executeForVO(  # pylint: disable=unexpected-keyword-arg
-                    vo, proxyUserName=proxyUser, proxyUserGroup=proxyGroup
-                )
-                if not res["OK"]:
-                    voRes[vo] = res["Message"]
+                # download a proxy and save a filename for future use:
+                result = getDNForUsername(proxyUser)
+                if not result["OK"]:
+                    self.log.error(f"Could not obtain a DN of user {proxyUser} for VO {vo}, skipped")
+                    continue
+                userDNs = result["Value"]  # a same user may have more than one DN
+                fd, filename = tempfile.mkstemp(prefix=vo + "__")
+                print("filename", filename)
+                os.close(fd)
+                vomsAttr = getVOMSAttributeForGroup(proxyGroup)
+                result = getProxy(userDNs, proxyGroup, vomsAttr=vomsAttr, proxyFilePath=filename)
```

Comment:
```suggestion
                with tempfile.NamedTemporaryFile(prefix=vo + "__") as ntf:
                    vomsAttr = getVOMSAttributeForGroup(proxyGroup)
                    result = getProxy(userDNs, proxyGroup, vomsAttr=vomsAttr, proxyFilePath=ntf.name)
```

Reply:
Well, according to the docs the file is deleted by default (`delete=True` by default, and even worse `delete_on_close` is True by default) We need the file for later (in` execute()`), so we need to keep it. The docs say that in case of `delete=False` `delete_on_close` is ignored (although not clear from the wording if the `with` statement has no precedence - have to check that).  I agree that the `with` statement looks nicer, at least it closes the file, but as I said above, we need the file - this is the whole idea of an early initialisation.

Reply:
OK, to preserve the file `delete=False` is required. It works with the `with` statement as well, it makes sense.

Reply:
Ah, I see. OK, use `delete=False` then.

Reply:
Done. I added a proxy renewal mechanism as well.